src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FeaturedTableService.cs

```diff
@@ -0,0 +1,232 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using GovUk.Education.ExploreEducationStatistics.Admin.Requests;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Admin.Validators;
+using GovUk.Education.ExploreEducationStatistics.Admin.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using static GovUk.Education.ExploreEducationStatistics.Admin.Validators.ValidationErrorMessages;
+using Unit = GovUk.Education.ExploreEducationStatistics.Common.Model.Unit;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class FeaturedTableService : IFeaturedTableService
+    {
+        private readonly ContentDbContext _contentDbContext;
+        private readonly IPersistenceHelper<ContentDbContext> _persistenceHelper;
+        private readonly IDataBlockService _dataBlockService;
+        private readonly IUserService _userService;
+        private readonly IMapper _mapper;
+
+        public FeaturedTableService(ContentDbContext contentDbContext,
+            IPersistenceHelper<ContentDbContext> persistenceHelper,
+            IDataBlockService dataBlockService,
+            IUserService userService,
+            IMapper mapper)
+        {
+            _contentDbContext = contentDbContext;
+            _persistenceHelper = persistenceHelper;
+            _dataBlockService = dataBlockService;
+            _userService = userService;
+            _mapper = mapper;
+        }
+
+        public async Task<Either<ActionResult, FeaturedTableViewModel>> Get(Guid releaseId, Guid dataBlockId)
+        {
+            return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+                .OnSuccess(_userService.CheckCanViewRelease)
+                .OnSuccess(async _ =>
+                    await _persistenceHelper.CheckEntityExists<FeaturedTable>(query =>
+                        query.Where(ft => ft.DataBlockId == dataBlockId)))
+                .OnSuccess(async featuredTable =>
+                {
+                    if (!await FeaturedTableIsAssociatedWithRelease(featuredTable, releaseId))
+                    {
+                        return new Either<ActionResult, FeaturedTableViewModel>(
+                            ValidationUtils.ValidationActionResult(FeaturedTableIsNotAssociatedWithRelease));
+                    }
```

Comment:
We don't need to throw a custom error as a 404 Not Found is sufficient to communicate that the featured table doesn't exist. We do this in many other places and there isn't anything special about this case.

It seems like we can essentially remove this in favour of simply:

```cs
.OnSuccess(async release =>
    await _persistenceHelper.CheckEntityExists<FeaturedTable>(query =>
        query.Where(ft => ft.DataBlockId == dataBlockId && ft.ReleaseId == release.id)))
.OnSuccess(featuredTable => _mapper.Map<FeaturedTableViewModel>(featuredTable));
```